### PR TITLE
Add user's unique email policy configuration

### PR DESCRIPTION
**Pull request by AdrienTorris**
By default, identity doesn't  require unique emails for user. 
Add an example of how configure ASP.NET Identity to have unique user emails, which is the need in a lot of projects.

### CODE REVIEW

File: aspnet/security/authentication/identity/sample/src/ASPNET-IdentityDemo/Startup.cs [hunk 1]
```diff
@@ -72,6 +72,9 @@ public void ConfigureServices(IServiceCollection services)
                 options.Cookies.ApplicationCookie.ExpireTimeSpan = TimeSpan.FromDays(150);
                 options.Cookies.ApplicationCookie.LoginPath = "/Account/LogIn";
                 options.Cookies.ApplicationCookie.LogoutPath = "/Account/LogOff";
+                
+                // User settings
+                options.User.RequireUniqueEmail = true;
```
Comment (Rick-Anderson) [inline aspnet/security/authentication/identity/sample/src/ASPNET-IdentityDemo/Startup.cs]: @HaoK  THIS seems like a good idea.

**Comment by Rick-Anderson** [comment]
@HaoK  one line change, please review.

File: aspnet/security/authentication/identity/sample/src/ASPNET-IdentityDemo/Startup.cs (hunk 1 above)
Comment (HaoK) [inline aspnet/security/authentication/identity/sample/src/ASPNET-IdentityDemo/Startup.cs]: Looks good to me

**Comment by Rick-Anderson** [comment]
@AdrienTorris  excellent. Thanks for doing this.

**Comment by AdrienTorris** [comment]
@Rick-Anderson Welcome, my pleasure ;)

**Pull request merged**